src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -15,9 +15,31 @@ public static class GarnetBuilderExtensions
     /// <summary>
     /// Adds a Garnet container to the application model.
     /// </summary>
-    /// <remarks>
-    /// The default image is "garnet" and the tag is "1.0.2".
-    /// </remarks>
+    /// <example>
+    /// <remarks>Use in AspireHost</remarks>
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api") 
+    /// var garnet = builder.AddGarnet("MyGarnet"); 
+    /// api.WithReference(garnet);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// <remarks>Use in Api</remarks>
+    /// <code>
+    /// var garnetConnect = Environment.GetEnvironmentVariable("services__garnet__0", EnvironmentVariableTarget.Process);
+    /// var host = garnetConnect[(garnetConnect.LastIndexOf('/') + 1)..garnetConnect.LastIndexOf(':')];
+    /// var port = int.Parse(garnetConnect[(garnetConnect.LastIndexOf(':') + 1)..]);
```

Comment:
We should remove this example. If this is what people need to do to use garnet then we should also write a component. I assumed people would use the redis component to interact with it.

Reply:
Fine. I change it to IConnectionMultiplexer.

Reply:
If we want to enable the garnet client as a primary client, then we need to solve the connection string problem here. This manual parsing should happen in the apphost, not in the user app.

Reply:
This was my problem in the example.
I put together a nuget pack, connected aspire.hosting.garnet
The connection string works correctly.

![image](https://github.com/dotnet/aspire/assets/52016832/2718d624-c146-4a9c-abb6-ea29f8090a1b)

![image](https://github.com/dotnet/aspire/assets/52016832/bce0dbe7-1023-46d8-a9df-2219436f64f3)

Reply:
Replaced it in the example with a connection string.
So do we use GarnetClient with the host and port highlighted from the connection line?
Or use redis connection?
```csharp
using var redis = await ConnectionMultiplexer.ConnectAsync(garnet);
            var db = redis.GetDatabase(0);
            await db.PingAsync();
            Console.WriteLine("RespPing: Success");
```

Reply:
Use the redis aspire component, which supports the connection string natively. Doesn't require extra parsing or splitting or anything.